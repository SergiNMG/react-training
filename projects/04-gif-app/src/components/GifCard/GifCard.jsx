import PropTypes from "prop-types";
import "./GifCard.scss";

export const GifCard = ({ title, text, image }) => {
  return (
    <>
      <div className="c-card">
        <img className="c-card__img" src={image} alt={text} />
        <p> {title} </p>
      </div>
    </>
  );
};

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
