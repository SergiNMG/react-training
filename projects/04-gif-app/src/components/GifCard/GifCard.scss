.c-card {
  display: flex;
  flex-direction: column;
  place-items: center;
  background-color: #444;
  border: 2px solid white;
  border-radius: 12px;

  &__img {
    border-radius: inherit;
  }
}
