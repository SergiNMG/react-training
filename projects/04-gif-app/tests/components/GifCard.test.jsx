import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard/GifCard.jsx";

describe("Test in GifCard.jsx", () => {

  const title = "goku";
  const image = "https://dragonball.com/goku.jpg";

  test("should match snapshot", () => {

    const { container } = render(<GifCard title={title} image={image} />);
    expect( container ).toMatchSnapshot();

  });

  test('should show the image with URL and ALT', () => { 

    render(<GifCard title={title} image={image} />);
    expect(screen.getByRole('img').src).toEqual(image);
  })

  test('should show the title in the component', () => {

    render(<GifCard title={title} image={image}/>);
    expect(screen.getByText( title )).toBeTruthy();

  })
});
