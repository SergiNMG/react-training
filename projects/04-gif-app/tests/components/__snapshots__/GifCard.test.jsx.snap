// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifCard.jsx should match snapshot 1`] = `
<div>
  <div
    class="c-card"
  >
    <img
      class="c-card__img"
      src="https://dragonball.com/goku.jpg"
    />
    <p>
       
      goku
       
    </p>
  </div>
</div>
`;
